packages/job-worker/src/playout/adlibUtils.ts

```diff
@@ -215,19 +214,9 @@ export async function innerStartQueuedAdLib(
 	// Ensure it is labelled as dynamic
 	newPartInstance.orphaned = 'adlib-part'
 
-	// Find the following part, so we can pick a good rank
-	const followingPart = selectNextPart(
-		context,
-		cache.Playlist.doc,
-		currentPartInstance,
-		null,
-		getOrderedSegmentsAndPartsFromPlayoutCache(cache),
-		false // We want to insert it before any trailing invalid piece
-	)
-	newPartInstance.part._rank = getRank(
-		currentPartInstance.part,
-		followingPart?.part?.segmentId === newPartInstance.segmentId ? followingPart?.part : undefined
-	)
+	// Parts are always integers spaced by one, and orphaned PartInstances will be decimals spaced between two Part
+	// so we can predict a 'safe' rank to get the desired position with some simple maths
+	newPartInstance.part._rank = getRank(currentPartInstance.part._rank, Math.floor(currentPartInstance.part._rank + 1))
```

Comment:
The documentation for the `Part._rank` property should be updated with this definition.